src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -0,0 +1,399 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "../../custom_node_interface.h"
+#include "nms.hpp"
+#include "opencv2/opencv.hpp"
+#include "utils.hpp"
+
+static constexpr const char* IMAGE_TENSOR_NAME = "image";
+static constexpr const char* SCORES_TENSOR_NAME = "scores";
+static constexpr const char* GEOMETRY_TENSOR_NAME = "geometry";
+static constexpr const char* TEXT_IMAGES_TENSOR_NAME = "text_images";
+static constexpr const char* COORDINATES_TENSOR_NAME = "text_coordinates";
+static constexpr const char* CONFIDENCE_TENSOR_NAME = "confidence_levels";
+
+int extract_text_images_into_output(struct CustomNodeTensor* output, const std::vector<cv::Rect>& boxes, const cv::Mat& originalImage, int targetImageHeight, int targetImageWidth, bool convertToGrayScale, const std::string& imageLayout) {
+    uint64_t outputBatch = boxes.size();
+    int channels = convertToGrayScale ? 1 : 3;
+
+    uint64_t byteSize = sizeof(float) * targetImageHeight * targetImageWidth * channels * outputBatch;
+
+    float* buffer = (float*)malloc(byteSize);
+
+    for (uint64_t i = 0; i < outputBatch; i++) {
+        cv::Size taretShape(targetImageWidth, targetImageHeight);
+        cv::Mat image = crop_and_resize(originalImage, boxes[i], taretShape);
+        if (convertToGrayScale) {
+            image = apply_grayscale(image);
+        }
+        if (imageLayout == "NCHW") {
+            auto imgBuffer = reorder_to_nchw((float*)image.data, image.rows, image.cols, image.channels());
+            std::memcpy(buffer + (i * channels * targetImageWidth * targetImageHeight), imgBuffer.data(), byteSize / outputBatch);
+        } else {
+            std::memcpy(buffer + (i * channels * targetImageWidth * targetImageHeight), image.data, byteSize / outputBatch);
+        }
+    }
+
+    output->data = reinterpret_cast<uint8_t*>(buffer);
+    output->dataBytes = byteSize;
+    output->dimsCount = 5;
+    output->dims = (uint64_t*)malloc(output->dimsCount * sizeof(uint64_t));
+    output->dims[0] = 1;
+    output->dims[1] = outputBatch;
+    if (imageLayout == "NCHW") {
+        output->dims[2] = channels;
+        output->dims[3] = targetImageHeight;
+        output->dims[4] = targetImageWidth;
+    } else {
+        output->dims[2] = targetImageHeight;
+        output->dims[3] = targetImageWidth;
+        output->dims[4] = channels;
+    }
+    output->precision = FP32;
+    return 0;
+}
+
+int extract_coordinates_into_output(struct CustomNodeTensor* output, const std::vector<cv::Rect>& boxes) {
```

Comment:
Also move to utils?

Reply:
This is very specific to this custom node and will not be reused in other nodes. Utils contain reusable components.

Reply:
Rename extract, this is not extracting

Reply:
Renamed